src/scheduler/FunctionCallClient.cpp

```diff
@@ -123,29 +127,35 @@ void FunctionCallClient::sendMPIMessage(
     if (faabric::util::isMockMode()) {
         mpiMessages.emplace_back(host, *msg);
     } else {
-        ClientContext context;
-        faabric::FunctionStatusResponse response;
-        CHECK_RPC("mpi_message", stub->MPICall(&context, *msg, &response));
+        SEND_MESSAGE_PTR(faabric::scheduler::FunctionCalls::MpiMessage, msg);
     }
 }
 
-faabric::HostResources FunctionCallClient::getResources(
-  const faabric::ResourceRequest& req)
+faabric::HostResources FunctionCallClient::getResources()
 {
+    faabric::ResponseRequest request;
     faabric::HostResources response;
-
     if (faabric::util::isMockMode()) {
         // Register the request
-        resourceRequests.emplace_back(host, req);
+        resourceRequests.emplace_back(host, request);
 
         // See if we have a queued response
         if (queuedResourceResponses[host].size() > 0) {
             response = queuedResourceResponses[host].dequeue();
         }
     } else {
-        ClientContext context;
-        CHECK_RPC("get_resources",
-                  stub->GetResources(&context, req, &response));
+        request.set_returnhost(faabric::util::getSystemConfig().endpointHost);
+
+        SEND_MESSAGE(faabric::scheduler::FunctionCalls::GetResources, request);
+
+        // Receive message
+        faabric::transport::Message msg =
+          awaitResponse(faabric::util::getSystemConfig().endpointHost,
+                        FUNCTION_CALL_PORT + REPLY_PORT_OFFSET);
+        // Deserialise message string
```

Comment:
I don't use `PARSE_RESPONSE` here because the response variable is defined in a larger scope (both for mock and non-mock mode). Alternatively, I could define and return from within both branches,